sys/net/ieee802154.txt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2021 Christian Amsüss <chrysn@fsfe.org>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_ieee802154 IEEE 802.15.4
+ * @ingroup     net
+ * @brief       IEEE 802.15.4 components
+ *
+ * General introduction
+ * --------------------
+ *
+ * The IEEE 802.15.4 standard describes radio communication for low-rate local
+ * wireless networks, operating in different frequency bands, modulations and
+ * MACs; all of these have been extended over time in revisions of the
+ * standard. It also describes topologies and roles of devices (eg. a PAN
+ * coordinator that is a FFD (Full Function Device) and RFDs (Reduced Function
+ * Devices) and clusters that use different PAN IDs to keep their traffic
+ * apart), without prescribing how they are formed or managed.
+ *
+ * Common frequencies used are the 868 MHz band (one channel, in Europe), the
+ * 915 MHz band (10 channels, in America) and the 2.4GHz band (16 channels);
+ * several more are defined between 169MHz and 9GHz.
+ *
+ * The standard describes several MACs that govern when to send and to listen;
+ * the choice of the MAC involves the trade-offs between power consumption,
+ * complexity and latency, as it governs how much time devices can spend with
+ * their radios powered down. Some MACs support channel hopping to avoid
+ * frequencies used by others. Common MACs are:
+ *
+ * * nonbeacon-eanbled: devices ask the coordinator to send a beacon in short
+ *   transmissions, then the coordinator sends a beacon guiding further
+ *   transmissions
+ * * beacon-enabled: a coordinator sends beacons regularly, devices can send
+ *   for some time after the beacon
+ * * TSCH: the network uses frequency and time slots synchronized at the
+ *   coordinator's beacons
+ * * DSME
+ *
+ * A comprehensive introduction to these is available in
+ * [Towards the Internet of Relevant Things: The IEEE 802.15.4e Standard](http://www.sigapp.org/sac/sac2016/T7-HandOut.pdf)
+ * presented Anastasi et al. at the ACM Symposium on Applied Computing 2016.
+ *
+ * Some MACs are established outside of the standard but either relevant for
+ * context or because they are used in RIOT:
+ * 
+ * * contikiMAC: receivers listen in brief time slots and stay listening if
+ *   they detect a carrier; senders send until they receive an acknowledgement.
+ * * "YOLO mode": receivers are always on; senders just send and hope for the
+ *   best.
+ * * "submac": YOLO mode, but with some common components of the specified MACs
+ *   implemented (Channel Clear Assessment (CCA), retransmissions,
+ *   acknowledgements)
```

Comment:
there's a gray zone here, because we actually never defined a "YOLO mode". Before the SubMAC radios would do what the driver implementor think it was best or what the radios provided as hardware acceleration. For instance, `at86rf2xx` and `mrf24j40` implement full CSMA-CA with retransmissions on hardware. Some radios only provided a hardware TX-ACK mechanism (`cc2538`, `cc2420`). Some others provide an ACK Timeout interrupt if there's no ACK (`kw41z`, `kw2xrf`). And there's the `nrf52840`, which didn't care about ACK at all.
Some of this radios were sending with CSMA-CA, other with CCA and some other with nothing at all.

Therefore I would say this mode is just an incomplete feature than a specific mode.

For the "submac YOLO mode", I think i would leave it either as "CSMA-CA senders" or simply indicate that it's an incomplete NBE (no discovery, no IT, etc).